### PR TITLE
[CSL-2225] Prettify `dbgen` code

### PR DESCRIPTION
**Pull request by ksaric**
https://iohk.myjetbrains.com/youtrack/issue/CSL-2225

@dcoutts We first require https://github.com/input-output-hk/cardano-sl/pull/2388 to be merged!

With this one merged, we can address https://github.com/input-output-hk/cardano-sl/pull/2400.

### CODE REVIEW

File: tools/src/dbgen/Lib.hs [hunk 1]
```diff
@@ -10,72 +10,138 @@ module Lib where
...
+          "none"  -> pure NoDistribution
+          "range" -> RangeDistribution <$> o .: "range" <*> o .: "amount"
+          _       -> fail ("Unknown type: " ++ distrType)
+
+instance ToJSON FakeUtxoCoinDistribution
+
+{-
+λ> decode $ "{\"type\":\"none\"}" :: Maybe FakeUtxoCoinDistribution
+Just NoDistribution
+λ> decode $ "{\"type\":\"range\",\"range\":1000}" :: Maybe FakeUtxoCoinDistribution
```
Comment (adinapoli-iohk) [inline tools/src/dbgen/Lib.hs]: I think these examples are now stale, as a `RangeDistribution` now takes also an `amount`.

**Comment by ksaric** [comment]
@adinapoli-iohk Tested, wallet generation worked on this PR. There may be an issue when creating a new wallet in some cases (I'm looking into it), but that's something I will address with a new fake tx generation PR.

**Comment by gromakovsky** [comment]
#2388 has been merged.

**Comment by adinapoli-iohk** [comment]
@ksaric There are conflicts now due to some our high-priority patches being merged -- would you mind fixing those? 😉

**Comment by ksaric** [comment]
@gromakovsky Can you merge this, please?

**Comment by gromakovsky** [comment]
@ksaric I can. The history looks a bit verbose though. Do you mind if I press `Squash and merge` button?

**Comment by ksaric** [comment]
@gromakovsky Yes, I agree with squashing.

**Pull request merged**